langchain/src/vectorstores/neo4jvector.ts

```diff
@@ -0,0 +1,573 @@
+  VECTOR = "vector",
+  HYBRID = "hybrid",
+}
+
+export enum DistanceStrategy {
+  EUCLIDEAN_DISTANCE = "euclidean",
+  COSINE = "cosine",
+}
+
+interface Neo4jVectorStoreArgs {
+  url: string;
+  username: string;
+  password: string;
+  database?: string;
+  preDeleteCollection?: boolean;
+  textNodeProperty?: string;
+  textNodeProperties?: string[];
+  embeddingNodeProperty?: string;
+  keywordIndexName?: string;
+  indexName?: string;
+  searchType?: SearchType;
+  retrievalQuery?: string;
+  nodeLabel?: string;
+  createIdIndex?: boolean;
+}
+
+const DEFAULT_SEARCH_TYPE = SearchType.VECTOR;
+const DEFAULT_DISTANCE_STRATEGY = DistanceStrategy.COSINE;
+
+export class Neo4jVectorStore extends VectorStore {
+  private driver: neo4j.Driver;
+
+  private database: string;
+
+  private preDeleteCollection: boolean;
+
+  private nodeLabel: string;
+
+  private embeddingNodeProperty: string;
+
+  private embeddingDimension: number;
+
+  private textNodeProperty: string;
+
+  private keywordIndexName: string;
+
+  private indexName: string;
+
+  private retrievalQuery: string;
+
+  private distanceStrategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY;
+
+  _vectorstoreType(): string {
+    return "neo4jvector";
+  }
+
+  constructor(embedings: Embeddings, config: Neo4jVectorStoreArgs) {
+    super(embedings, config);
+  }
+
+  static async initialize(embedings: Embeddings, config: Neo4jVectorStoreArgs) {
+    const store = new Neo4jVectorStore(embedings, config);
+    await store._initializeDriver(config);
+    await store._verifyConnectivity();
+
+    const {
+      preDeleteCollection = false,
+      nodeLabel = "Chunk",
+      textNodeProperty = "text",
+      embeddingNodeProperty = "embedding",
+      keywordIndexName = "keyword",
+      indexName = "vector",
+      retrievalQuery = "",
+    } = config;
+
+    store.embeddingDimension = (await embedings.embedQuery("foo")).length;
+    store.preDeleteCollection = preDeleteCollection;
+    store.nodeLabel = nodeLabel;
+    store.textNodeProperty = textNodeProperty;
+    store.embeddingNodeProperty = embeddingNodeProperty;
+    store.keywordIndexName = keywordIndexName;
+    store.indexName = indexName;
+    store.retrievalQuery = retrievalQuery;
+
+    if (store.preDeleteCollection) {
+      await store._dropIndex();
+    }
+
+    return store;
+  }
+
+  async _initializeDriver({
+    url,
+    username,
+    password,
+    database = "neo4j",
+  }: Neo4jVectorStoreArgs) {
+    try {
+      this.driver = neo4j.driver(url, neo4j.auth.basic(username, password));
+      this.database = database;
+    } catch (error) {
+      throw new Error(
+        "Could not create a Neo4j driver instance. Please check the connection details."
+      );
+    }
+  }
+
+  async _verifyConnectivity() {
+    try {
+      const session = this.driver.session({ database: this.database });
+      await session.close();
+    } catch (error: any) {
+      console.log("Failed to verify connection.");
+    }
+  }
```

Comment:
This would be preferred https://github.com/langchain-ai/langchainjs/blob/efc16f84f94bcea225fb9f306f7db0c59ef03d24/langchain/src/graphs/neo4j_graph.ts#L72-L74